resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/stream/config/AkkaRunnableGraph.java

```diff
@@ -0,0 +1,132 @@
+
+    @Bean
+    KafkaProducerService kafkaProducerService(final Sink<ProducerRecord<String, ResourceRequest>, CompletionStage<Done>> sink,
+                                              final ConsumerTopicConfiguration upstreamConfig,
+                                              final Materializer materializer) {
+        return new KafkaProducerService(sink, upstreamConfig, materializer);
+    }
+
+    @Bean
+    Function1<Throwable, Directive> supervisor() {
+        return ex -> Supervision.resumingDecider().apply(ex);
+    }
+
+    @Bean
+    RunnableGraph<DrainingControl<Done>> runner(
+            final Source<CommittableMessage<String, ResourceRequest>, Control> source,
+            final Sink<Envelope<String, ResourceResponse, Committable>, CompletionStage<Done>> sink,
+            final Function1<Throwable, Directive> supervisionStrategy,
+            final ProducerTopicConfiguration topicConfiguration,
+            final StreamingResourceServiceProxy service) {
+        // Get output topic from config
+        Topic outputTopic = topicConfiguration.getTopics().get("output-topic");
+
+        // Read messages from the stream source
+        return source
+                .map(message -> new Pair<>(message, Optional.ofNullable(message.record().value())))
+                // Get a stream of resources from an iterator
+                .flatMapConcat(messageAndRequest -> messageAndRequest.second()
+                        // If the request optional is not empty then we need to process the message
+                        // Get a stream of resources from the implemented resource-service
+                        .map(request -> Source.fromIterator(() -> service.getResourcesById(request.resourceId))
+                                // Make the stream of resources an Optional
+                                .map(Optional::of)
+                                // Add empty optional to the end of the stream so that we can identify
+                                // when we have processed the last resource in the stream.
+                                // This will allow us to then acknowledge the original upstream message
+                                .concat(Source.single(Optional.empty()))
+                                // Build the producer record for each leaf resource within the Optional
+                                .map(resourceOptional -> resourceOptional
+                                        .map(leafResource -> new ProducerRecord<>(
+                                                outputTopic.getName(),
+                                                messageAndRequest.first().record().partition(),
+                                                messageAndRequest.first().record().key(),
+                                                Builder.create(messageAndRequest.first().record().value()).withResource(leafResource),
+                                                messageAndRequest.first().record().headers())
```

Comment:
This could look a little tidier pulling out the `ConsumerRecord record = messageAndRequest.first().record()` and using that here